Batch edit attributes

steff-o opened the pull request:
Solves #1316. Makes it possible to batch edit attributes using the normal attribute editor form.

To batch edit attributes, the attribute must be configured with `"allowBatchEdit": true` in the _attributes_ section of the layer configuration.

If more than one feature is selected when the attribute edit form is opened, there will be a checkbox for each attribute that can be changed. Upon completion, all selected features will receive the new value for the attributes that have been checked.

To select multiple features use shift-click in the map.

johnnyblasta commented:
Tested and it works as expected editing in batch.
However there are som errors introduced.

1. When only one feature is edited this error is thrown:
 ```
Uncaught TypeError: Cannot read properties of null (reading 'addEventListener')
    at Object.fn [as addListener] (edithandler.js:907)
```
2. When no attribute has `"allowBatchEdit": true` and multiple features are selected this error is thrown:
```
Uncaught TypeError: Cannot read properties of undefined (reading 'reduce')
    at editAttributes (edithandler.js:1076)
```

Also there is some lints in edithandler.js with comments and let instead of const.

steff-o commented:
Great testing. Apparently I forgot to check some of the basic cases. I'll have a look.

steff-o commented:
I have fixed the problem where no attribute is configured for batch edit, but I can not reproduce the first error. It seems to be related to when constraints are in place as that function is only called when using a constraint. But the only error I can get related to constraints is when an attribute with a constraint has `allowBatchEdit` but the attribute that it depends on does not and you select at least two objects. But that seems to me to be missconfigured.

While digging in the constraints I changed so that if a batch edited attribute is changed, its dependent attributes are also always changed without a checkbox to keep consistent with non-batch edits.

Also fixed some lints.
